packages/wallet/src/services/DRepRegistrationTracker.ts

```diff
@@ -1,31 +1,54 @@
+/* eslint-disable sonarjs/cognitive-complexity */
+import * as Crypto from '@cardano-sdk/crypto';
 import { Cardano } from '@cardano-sdk/core';
-import { Observable, map } from 'rxjs';
+import { Observable, from, map, mergeMap } from 'rxjs';
 import { TrackerSubject } from '@cardano-sdk/util-rxjs';
 
 export interface CreateDRepRegistrationTrackerProps {
   historyTransactions$: Observable<Cardano.HydratedTx[]>;
+  getPubDRepKey: () => Promise<Crypto.Ed25519PublicKeyHex | undefined>;
 }
 
+export interface IsOwnDRepCreddentialProps {
+  certificate: Cardano.Certificate;
+  dRepKeyHash: Crypto.Ed25519KeyHashHex;
+}
+
+const hasOwnDRepCredential = ({ certificate, dRepKeyHash }: IsOwnDRepCreddentialProps) =>
+  'dRepCredential' in certificate &&
+  certificate.dRepCredential.type === Cardano.CredentialType.KeyHash &&
+  certificate.dRepCredential.hash === Crypto.Hash28ByteBase16.fromEd25519KeyHashHex(dRepKeyHash);
+
+const getDRepKeyHash = async (dRepKey: Crypto.Ed25519PublicKeyHex | undefined) =>
+  dRepKey ? (await Crypto.Ed25519PublicKey.fromHex(dRepKey).hash()).hex() : undefined;
+
 export const createDRepRegistrationTracker = ({
-  historyTransactions$
+  historyTransactions$,
+  getPubDRepKey
 }: CreateDRepRegistrationTrackerProps): TrackerSubject<boolean> =>
   new TrackerSubject(
-    historyTransactions$.pipe(
-      map((txs) => {
-        const reverseTxs = txs.reverse();
+    from(getPubDRepKey().then(getDRepKeyHash)).pipe(
+      mergeMap((dRepKeyHash) =>
+        historyTransactions$.pipe(
+          map((txs) => {
+            if (!dRepKeyHash) return false;
+            const reverseTxs = [...txs].reverse();
 
-        for (const {
-          body: { certificates }
-        } of reverseTxs) {
-          if (certificates) {
-            for (const certificate of certificates) {
-              if (certificate.__typename === Cardano.CertificateType.UnregisterDelegateRepresentative) return false;
-              if (certificate.__typename === Cardano.CertificateType.RegisterDelegateRepresentative) return true;
+            for (const {
+              body: { certificates }
+            } of reverseTxs) {
+              if (certificates) {
+                for (const certificate of certificates) {
+                  if (!hasOwnDRepCredential({ certificate, dRepKeyHash })) continue;
```

Comment:
I like the separate refactor commit, but it includes fixes to the 1st commit. Please either squash the 2 commits into 1, or hoist the new feature fixes from the `refactor` commit to the `feat` one.

Reply:
done